src/solver/conv_ocl_dir2Dfwd_exhaustive_search.cpp

```diff
@@ -447,70 +429,99 @@ ConvOclDirectFwdLegacyExhaustiveSearch::SearchImpl(const ConvolutionContext& par
-                            ++failed_counter;
-                            continue;
+                            auto solution = solver.GetSolution(params, result);
+                            std::ostringstream os_params;
+                            result.Serialize(os_params);
+                            solution.performance_config = os_params.str();
+                            all_solutions.push_back(solution);
                         }
-
-                        is_passed = true;
-                        MIOPEN_LOG_T("##(n_current, n_failed, n_runs_total): " << run_counter
-                                                                               << " / "
-                                                                               << failed_counter
-                                                                               << " / "
-                                                                               << total_runs
-                                                                               << " elapsed_time: "
-                                                                               << processing_time
-                                                                               << " best_time: "
-                                                                               << processing_time
-                                                                               << ", "
-                                                                               << result);
-
-                        if(processing_time < min_proc_time)
-                        {
-                            MIOPEN_LOG_I('#' << run_counter << ' ' << processing_time << " < "
-                                             << min_proc_time
-                                             << ' '
-                                             << result);
-                            min_proc_time = processing_time;
-                            candidate     = result;
-                        }
-
-                        if(run_counter % report_inteval == 0)
-                        {
-                            MIOPEN_LOG_W("Runs left: " << runs_left << ", "
-                                                       << "min time so far: "
-                                                       << min_proc_time
-                                                       << ", "
-                                                       << "curr time: "
-                                                       << processing_time
-                                                       << ' '
-                                                       << result);
-                        }
-                        run_counter++;
                     }
                 }
             }
         }
+// PrecompileKernels call saves to binary_cache, this needs to be escaped if KERN_CACHE is not on.
+#if MIOPEN_ENABLE_SQLITE_KERN_CACHE
```

Comment:
Please use this to decide is binary cache is enabled (rename to `IsBinaryCacheDisabled` and make it public):
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/584da20a32d26164ac1b65966eda54a1524a6801/src/binary_cache.cpp#L103

Reply:
Whether binary  cache is enabled or disabled, parallel compilation by `PrecompileKernels`  is required here. And `LoadProgram` called in `PrecompileKernels` has the mechanism of whether binary cache is disabled or not. 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8fe06e67a16b7261d59d70a0ae75d83e678ad788/src/hip/handlehip.cpp#L411-L421
So I think there is no need to judge the binary cache is disabled or not here. And This also applies to `generic_search`:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8fe06e67a16b7261d59d70a0ae75d83e678ad788/src/include/miopen/generic_search.hpp#L352-L365
According to the current situation, if the binary cache is disabled, then MIOpen will still perform serial compilation here and in `generic_search`, this is not what we want.

Reply:
Do I need to incorporate this "search" code into `generic_search`? I'm not sure I can do it as your wish.

Reply:
Updates to the generic_search should be applied in a separate PR.